docs/container.md

```diff
@@ -0,0 +1,119 @@
+<!--
+SPDX-FileCopyrightText: Red Hat, Inc.
+SPDX-License-Identifier: GPL-2.0-or-later
+-->
+
+# Using the ovirt-imageio container
+
+## Overview
+
+You can use ovirt-imageio to transfer images without oVirt using the
+ovirt-imageio container image.
+
+This way, you can run an image container for every transfer,
+exposing the container for remote access, and connecting the
+relevant storage. When the transfer is done, the user will
+unexpose the container and terminate it.
+
+In this mode, instead of having to add a ticket using the control
+service, the imageio server will serve a single ticket, automatically
+generated in the container when starting.
+Actually, the control service is disabled.
+
+To obtain the public container image you can run:
+
+    podman pull quay.io/ovirt/ovirt-imageio
+
+## Run the container
+
+The ovirt-imageio container have an entrypoint script that takes
+additional arguments:
+
+```
+$ podman run \
+        --interactive \
+        --tty \
+        --rm \
+        quay.io/ovirt/ovirt-imageio:latest --help
+usage: entrypoint.py [-h] [--ticket-id TICKET_ID] image
+
+ovirt-imageio
+
+positional arguments:
+  image                 Path to image.
+
+options:
+  -h, --help            show this help message and exit
+  --ticket-id TICKET_ID
+                        Optional. Set the ID of the ticket.
+                        It is recommended to use with random string for better security.
+                        Defaults to the image filename.
+```
+
+To run the container and serve an image, you need to mount the
+image in the container, and pass the full path to the image
+inside the container in the command line.
+
+The container script will explore the image properties, create a ticket for
+you, serve the image using an NBD server, and expose it through http at
+the port 80 using the ovirt-imageio server.
+
+### Example session
+
+Create the image that we want to expose through the
+containerized ovirt-imageio:
+
+```
+    $ mkdir /var/tmp/images
+    $ qemu-img create -f raw /var/tmp/images/disk.raw 6g
+```
+
+Load the file ticket in the examples folder.
+
+```
+    $ podman run \
+        --interactive \
+        --tty \
+        --rm \
+        --publish 8080:80 \
+        --volume /var/tmp/images:/image:Z \
+        quay.io/ovirt/ovirt-imageio:latest --ticket-id example /image/disk.raw
+    2022-10-08 14:41:14,961 INFO    (MainThread) [server] Starting (hostname=51365cc84a8b pid=1, version=2.4.7)
+    2022-10-08 14:41:14,966 INFO    (MainThread) [services] remote.service listening on ('::', 80)
+    2022-10-08 14:41:14,966 INFO    (MainThread) [server] Initial ticket: /app/ticket.json
+    2022-10-08 14:41:14,967 INFO    (MainThread) [server] Ready for requests
+```
+
+In another shell, download the image (e.g., with `curl`):
+```
+    $ curl http://localhost:8080/images/example -o /data/tmp/download.raw
```

Comment:
`/data/tmp/` is from my system, this may confuse the user. Use `/var/tmp/download.raw`?